apps/demo-pkp-social-auth-next-ts/src/pages/index.tsx

```diff
@@ -0,0 +1,496 @@
+  async function handleConnectWallet(c) {
+    const { account, chain, connector } = await connectAsync(c);
+    try {
+      await authWithWallet(account, connector);
+    } catch (err) {
+      console.error(err);
+      setError(err);
+      setView(Views.ERROR);
+    }
+  }
+
+  async function authWithWallet(address, connector) {
+    setView(Views.REQUEST_AUTHSIG);
+
+    // Get auth sig
+    const signer = await connector.getSigner();
+    const signAuthSig = async (message: string) => {
+      const sig = await signer.signMessage(message);
+      return sig;
+    };
+    const authMethod = await litAuthClient.signInWithEthWallet({
+      address,
+      signMessage: signAuthSig,
+    });
+    setAuthMethod(authMethod);
+    console.log('authMethod', authMethod);
+
+    // Fetch PKPs associated with eth wallet account
+    setView(Views.FETCHING);
+    const pkps: IRelayPKP[] = await litAuthClient.fetchPKPsByAuthMethod(
+      authMethod
+    );
+    if (pkps.length > 0) {
+      console.log(pkps);
+      setPKPs(pkps);
+    }
+    setView(Views.FETCHED);
+  }
+
+  /**
+   * Handle redirect from Lit login server
+   */
+  const handleRedirect = useCallback(async () => {
+    setView(Views.HANDLE_REDIRECT);
+    try {
+      // Get auth method object that has the OAuth token from redirect callback
+      const authMethod: AuthMethod = litAuthClient.handleSignInRedirect();
+      setAuthMethod(authMethod);
+
+      // Fetch PKPs associated with social account
+      setView(Views.FETCHING);
+      const pkps: IRelayPKP[] = await litAuthClient.fetchPKPsByAuthMethod(
+        authMethod
+      );
+      if (pkps.length > 0) {
+        setPKPs(pkps);
+      }
+      setView(Views.FETCHED);
+    } catch (err) {
+      console.error(err);
+      setError(err);
+      setView(Views.ERROR);
+    }
+
+    // Clear url params once we have the OAuth token
+    // Be sure to use the redirect uri route
+    router.replace(window.location.pathname, undefined, { shallow: true });
+  }, [litAuthClient, router]);
+
+  /**
+   * Mint a new PKP for current auth method
+   */
+  async function mint() {
+    setView(Views.MINTING);
+
+    try {
+      // Mint new PKP
+      const newPKP: IRelayPKP = await litAuthClient.mintPKPWithAuthMethod(
+        authMethod
+      );
+
+      // Add new PKP to list of PKPs
+      const morePKPs: IRelayPKP[] = [...pkps, newPKP];
+      setPKPs(morePKPs);
+
+      setView(Views.MINTED);
+      setView(Views.CREATING_SESSION);
+
+      // Get session sigs for new PKP
+      await createSession(newPKP);
+    } catch (err) {
+      console.error(err);
+      setError(err);
+      setView(Views.ERROR);
+    }
+  }
+
+  /**
+   * Generate session sigs for current PKP
+   *
+   * @param {Object} pkp - PKP object
+   */
+  async function createSession(pkp: IRelayPKP) {
+    setView(Views.CREATING_SESSION);
+
+    let authNeededCallback;
+    try {
+      if (authMethod.authMethodType === AuthMethodType.EthWallet) {
```

Comment:
it would be great, if possible, to not have the dapp care / implement this if / else and we just implement it for them in a single `getAuthNeededCallback` call (and we do the if/else inside our SDK). Hell, it'd be cool to have the dapp not make this call at all. Might mean the SDK maintains more state. wdyt?